src/server/deposit_claim_txhash.rs

```diff
@@ -0,0 +1,66 @@
+use std::sync::Arc;
+
+use crate::state::database::DatabaseExt;
+use crate::state::{AppState, DatabaseError};
+use crate::try_start_session;
+use axum::extract::{Query, State};
+use axum::response::IntoResponse;
+use axum::Json;
+use axum_auto_routes::route;
+use mongodb::bson::doc;
+use reqwest::StatusCode;
+use serde::{Deserialize, Serialize};
+
+use super::responses::{ApiResponse, Status};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DepositClaimTxhashQuery {
+    btc_txid: String,
+}
+
+#[route(get, "/deposit_claim_txhash")]
+pub async fn deposit_claim_txhash(
+    State(state): State<Arc<AppState>>,
+    Query(query): Query<DepositClaimTxhashQuery>,
+) -> impl IntoResponse {
+    let mut session = try_start_session!(state);
+    if let Err(err) = session.start_transaction().await {
```

Comment:
No need to start a transaction here because we only perform one read. It's useful when we do some writing where order is important. Also make sure to close a transaction if you start one (for session it is not need though, it should be dropped automatically when session goes out of scope).